server/app/repository/ReportingRepository.java

```diff
@@ -21,11 +20,19 @@ public final class ReportingRepository {
 
   private final Clock clock;
   private final Database database;
+  private final ImmutableList<ProgramModel> listOfPrograms;
 
-  @Inject
-  public ReportingRepository(Clock clock) {
+  public ReportingRepository() {
+    this.clock = null;
+    this.database = null;
+    this.listOfPrograms = null;
+  }
```

Comment:
Guice will throw a MISSING_CONSTRUCTOR error when it can't find an [injectable constructor](https://github.com/google/guice/JustInTimeBindings#inject-constructors) to use to create the requested objects.
https://github.com/google/guice/wiki/MISSING_CONSTRUCTOR

Is there a flag that opts-out this class?

Reply:
@zarifzm that's likely due to attempting to inject it directly somewhere instead of using the factory you created. Check the rest of the code base to see if you can find another class with a c'tor annotated with `@Inject` that takes `ReportingRepository`, change it to use the factory method and the error should go away.